server/internal/core/application/utils.go

```diff
@@ -118,6 +119,8 @@ func (m *txRequestsQueue) pop(num int64) ([]domain.TxRequest, []ports.BoardingIn
 		}
 		// Skip tx requests for which users didn't notify to be online in the last minute.
 		if p.pingTimestamp.IsZero() || time.Since(p.pingTimestamp).Minutes() > 1 {
+			log.Debugf("delete tx request %s because it didn't ping in the last minute", p.Id)
+			delete(m.requests, p.Id) // cleanup the request from the map
```

Comment:
let's add an if and delete form the queue only if the last ping was longer than 5 minutes ago. Also, add a TODO to move this to a dedicated func. It's ok for now to have a side effect but we don't really want to keep it like this.

```suggestion
			// TODO: move to dedicated function.
			if p.pingTimestamp.IsZero() || time.Since(p.pingTimestamp).Minutes() > 1 {
			         log.Debugf("delete tx request %s because it didn't ping in the last minute", p.Id)
			         delete(m.requests, p.Id) // cleanup the request from the map
			}
```